scripts/post_deploy_sns.py

```diff
@@ -0,0 +1,106 @@
+delete the subscription, regardless if the subscription was for a group email
+address.
+"""
+import logging
+
+from furl import (
+    furl,
+)
+
+from azul import (
+    config,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+
+def main() -> None:
+    if config.disable_monitoring:
+        print('SNS subscription confirmation skipped, monitoring is disabled.')
+    elif subscription_pending_confirmation():
+        url = prompt_for_subscription_url()
+        confirm_subscription(url)
+    else:
+        print('SNS subscription confirmation skipped, no pending subscriptions found.')
+
+
+def sns_topic_arn() -> str:
+    """
+    Return the ARN of the SNS topic.
+    """
+    return ':'.join(
+        (
+            'arn',
+            'aws',
+            'sns',
+            config.region,
+            config.aws_account_id,
+            config.sns_topic_qualified_name
+        )
+    )
+
+
+def subscription_pending_confirmation() -> bool:
+    """
+    Return True if there is a pending subscription to the deployment's SNS topic.
+    """
+    paginator = aws.sns.get_paginator('list_subscriptions_by_topic')
+    pending_subs = []
+    for page in paginator.paginate(TopicArn=sns_topic_arn()):
+        for subscription in page['Subscriptions']:
+            if subscription['SubscriptionArn'] == 'PendingConfirmation':
+                assert config.azul_notification_email == subscription['Endpoint'], subscription
+                pending_subs.append(subscription)
+    if pending_subs:
+        assert 1 == len(pending_subs), pending_subs
+        return True
+    else:
+        return False
+
+
+def prompt_for_subscription_url() -> furl:
+    """
+    Prompt user for a confirmation URL.
+    """
+    print('\n'
+          f'To confirm the pending subscription of {config.azul_notification_email}\n'
+          'to the SNS topic that receives monitoring and security notifications, please\n'
+          'open the "Subscription Confirmation" email, copy the "Confirm subscription"\n'
+          'link address, and paste the value here.\n')
+    while True:
+        url = input('URL: ').strip()
+        if url:
+            return furl(url)
+
+
+def confirm_subscription(url: furl) -> None:
+    """
+    Confirm a SNS subscription and enable AuthenticateOnUnsubscribe.
+    """
+    topic_arn = url.args['TopicArn']
+    token = url.args['Token']
+    endpoint = url.args['Endpoint']
+
+    response = aws.sns.confirm_subscription(TopicArn=topic_arn,
+                                            Token=token,
+                                            AuthenticateOnUnsubscribe='true')
+    if response and response['SubscriptionArn'].startswith('arn:'):
+        print('Subscription confirmed for:', endpoint)
+    else:
+        raise RuntimeError('Unexpected response while confirming SNS subscription',
```

Comment:
Have you ever observed this branch to be hit?

Reply:
Only during development when I ended up with multiple subscriptions pending a confirmation.
I will remove the `if/else` and assume that if `aws.sns.confirm_subscription()` completes without raising an error,  the subscription has been confirmed.